analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LoopsAndLinq.CSharp11.cs

```diff
@@ -1,37 +1,99 @@
 ﻿using System;
 using System.Collections.Generic;
 
-namespace Tests.Diagnostics
+class IsPatternTests
 {
-    class IsPatternTests
+    void ListPattern(List<int[]> list)
     {
-        void ListPattern(List<int[]> list)
+        foreach (int[] array in list) // Noncompliant
         {
-            foreach (int[] array in list) // Noncompliant
+            if (array is [1, 2, 3]) // Secondary
             {
-                if (array is [1, 2, 3]) // Secondary
-                {
-                    Console.WriteLine("Pattern match successful");
-                }
+                Console.WriteLine("Pattern match successful");
             }
+        }
 
-            foreach (var array in list) // Compliant, do not raise on VarPattern in ListPattern
+        foreach (var array in list) // Compliant, do not raise on VarPattern in ListPattern
+        {
+            if (array is [1, var x, var z])
             {
-                if (array is [1, var x, var z])
-                {
-                    Console.WriteLine("Pattern match successful");
-                }
-
+                Console.WriteLine("Pattern match successful");
             }
 
-            foreach (var array in list) // Compliant, do not raise on declaration statements in ListPattern 
-            {
-                if (array is [1, ..] local)
-                {
-                    Console.WriteLine("Pattern match successful");
-                }
+        }
 
+        foreach (var array in list) // Compliant, do not raise on declaration statements in ListPattern
+        {
+            if (array is [1, ..] local)
+            {
+                Console.WriteLine("Pattern match successful");
             }
+
+        }
+    }
+}
+
+// https://github.com/SonarSource/sonar-dotnet/issues/7730
+class Repro_7730
```

Comment:
IMO we should split this into two different issues/reproducers:
1) 7730: FP - object being iterated should be IEnumerable/IQueryable
2) ????: FN - logical pattern matching/ logical operators

Why?
They require 2 different fixes that are not related to each other.